baseapp/auth/saml/state.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package saml
+
+import (
+	"net/http"
+	"time"
+)
+
+// IDStore stores the request id for SAML auth flows
+type IDStore interface {
+	// StoreID stores a request ID in such a way that it can be
+	// retreived later using GetIDs
+	StoreID(w http.ResponseWriter, id string) error
```

Comment:
This should take the `http.Request` as well for implementations that need information from the request context to access the session store.